core/src/main/java/com/graphhopper/routing/util/parsers/OSMHazmatTunnelParser.java

```diff
@@ -0,0 +1,42 @@
+package com.graphhopper.routing.util.parsers;
+
+import java.util.List;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.profiles.EncodedValue;
+import com.graphhopper.routing.profiles.EncodedValueLookup;
+import com.graphhopper.routing.profiles.EnumEncodedValue;
+import com.graphhopper.routing.profiles.HazmatTunnel;
+import com.graphhopper.routing.util.EncodingManager.Access;
+import com.graphhopper.storage.IntsRef;
+
+public class OSMHazmatTunnelParser implements TagParser {
+    
+    private final EnumEncodedValue<HazmatTunnel> hazTunnelEnc;
+    
+    public OSMHazmatTunnelParser() {
+        this.hazTunnelEnc = new EnumEncodedValue<>(HazmatTunnel.KEY, HazmatTunnel.class);
+    }
+
+    @Override
+    public void createEncodedValues(EncodedValueLookup lookup,
+                    List<EncodedValue> registerNewEncodedValue) {
+        registerNewEncodedValue.add(hazTunnelEnc);
+    }
+
+    @Override
+    public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay readerWay, Access access,
+                    long relationFlags) {
+        if (readerWay.hasTag("hazmat:E", "no")) {
+            hazTunnelEnc.setEnum(false, edgeFlags, HazmatTunnel.E);
+        } else if (readerWay.hasTag("hazmat:D", "no")) {
+            hazTunnelEnc.setEnum(false, edgeFlags, HazmatTunnel.D);
+        } else if (readerWay.hasTag("hazmat:C", "no")) {
+            hazTunnelEnc.setEnum(false, edgeFlags, HazmatTunnel.C);
+        } else if (readerWay.hasTag("hazmat:B", "no") || readerWay.hasTag("hazmat:A", "no")) {
+            hazTunnelEnc.setEnum(false, edgeFlags, HazmatTunnel.B);
```

Comment:
This looks a bit illogical to me. That it has the value "B" although B is not allowed.

Reply:
You're right. I misinterpreted the meaning of A because it seemed a bit like a special case to me.